code/Pagetypes/FreshdeskPage.php

```diff
@@ -0,0 +1,262 @@
+        }
+
+        $tickets = $this->getTickets($currentMember);
+
+        if (empty($tickets) || 0 == count($tickets)) {
+            return false;
+        }
+
+        $productID = false;
+        if (defined('FRESHDESK_PRODUCT_ID')) {
+            $productID = FRESHDESK_PRODUCT_ID;
+        }
+
+        // Get all open by default
+        $filter = ['status' => 2];
+        if ($this->request->getVar('status')) {
+            $filter['status'] = $this->request->getVar('status');
+        }
+        if ($this->request->getVar('priority')) {
+            $filter['priority'] = $this->request->getVar('priority');
+        }
+        $filter = $this->validateFilter($filter);
+
+        $tickets = $this->freshdeskTicketFilter($tickets, $productID, $filter);
+        $tickets = $this->humanReadable($tickets);
+        $tickets = new ArrayList($tickets);
+
+        return new PaginatedList($tickets, $this->request);
+    }
+
+    /**
+     * Renders a form which can be used for filtering the tickets via template.
+     *
+     * @return Form $form
+     */
+    public function filterForm()
+    {
+        $statuses = Config::inst()->get('FreshdeskPage_Controller', 'freshdeskStatus');
+        $currentStatus = 2;
+        if ($this->request->getVar('status') && null != $this->request->getVar('status')) {
+            $currentStatus = $this->request->getVar('status');
+        }
+
+        $priorities = Config::inst()->get('FreshdeskPage_Controller', 'freshdeskPriority');
+        $priorities[0] = 'any';
+        $currentPriority = 0;
+        if ($this->request->getVar('priority') && null != $this->request->getVar('priority')) {
+            $currentPriority = $this->request->getVar('priority');
+        }
+
+        $fields = new FieldList(
+            DropdownField::create('status', 'status', $statuses, $currentStatus),
+            DropdownField::create('priority', 'priority', $priorities, $currentPriority)
+        );
+
+        $actions = new FieldList(
+            FormAction::create('Filter', 'Filter')
+        );
+
+        $form = new Form($this, '', $fields, $actions);
+        $form->setTemplate('FilterForm');
+        $form->setFormMethod('get');
+
+        return $form;
+    }
+
+    /**
+     * Get tickets from in memory cache, otherwise put them there.
+     *
+     * @param mixed $currentMember
+     *
+     * @return array $_tickets
+     */
+    private function getTickets($currentMember)
+    {
+        if (!self::$_tickets) {
+            $freshdesk = \FreshdeskAPI::create();
+            $result = $freshdesk->APICall('GET', FRESHDESK_API_BASEURL, '/api/v2/tickets?email='.urlencode($currentMember->Email));
+
+            $tickets = [];
+            if ($result && '200' == $result->getStatusCode()) {
+                $tickets = json_decode($result->getBody()->getContents(), true);
+            }
+            self::$_tickets = $tickets;
+        }
+
+        return self::$_tickets;
```

Comment:
Might be good idea to get this to return `ArrayList` - keeps the abstractions tidy by making this gtetter work similar to DataObject::get

Reply:
ack